libs/partners/groq/langchain_groq/chat_models.py

```diff
@@ -511,19 +512,19 @@ def _stream(
             generation = chat_result.generations[0]
             message = cast(AIMessage, generation.message)
             tool_call_chunks = [
-                {
-                    "name": rtc["function"].get("name"),
-                    "args": rtc["function"].get("arguments"),
-                    "id": rtc.get("id"),
-                    "index": rtc.get("index"),
-                }
+                create_tool_call_chunk(
```

Comment:
There are places we've duplicated logic (e.g., in `astream`) that may also need to be updated.

What will the behavior be if a tool call is missing a `type`? Possible there will be issues operating on old serialized tool calls (ok if so-- helps to know in case bugs pop up)? And do we need to update other integrations (e.g., community and other repos)?

Reply:
only thing that depends on type atm is Tools when they're taking a ToolCall (which is a new feature, you couldn't pass in ToolCalls at all before)